.gitignore

```diff
@@ -0,0 +1,5 @@
+# Object file
+*.o
+
+# ignore .md files
+test.mdnode_modules/
```

Comment:
I suggest you put the `test.md` and `node_modules` in separate lines. 

This is because when trying to ignore the listed files, git will see the current listing (i.e. `test.mdnode_modules/`) as a single directory name. Therefore, you'll still end up committing the `test.md` and `node_modules` to git history. Please list them in different lines. E.g.

```
test.md
/node_modules
```